### DISCUSSION TITLE
✅ Adding Private key file using iTunes file sharing

### DISCUSSION THREAD

**Discussion by bellaf** [isontheline/pro.webssh.net#243] [Q&A] [answered]
I am trying to use my private key file with this app (I have been using it for years and love it BTW, but without key files)..
I have added the private, OpenSSH, keyfile to the webssh Files section in iTunes, and Sync'd the phone, but when I try and add a Key file, it doesnt list it, or allow me to select it in any way.  I am only able to "paste" or type in (eek!) the Private key data.
Any help would be gratefully received..

**Comment by isontheline** [comment, marked as answer]
Hello @bellaf 👋

Thanks for using WebSSH!

You're right there is a bug : Private key files aren't imported

I'm opening an issue right now : https://github.com/isontheline/pro.webssh.net/issues/244

**Comment by bellaf** [comment]
Many thanks, and thanks for the speedy response 😁🤣.

Let me know if can test something ...

Cheers

Tony Bell

**Reply by isontheline** [reply]
Hello @bellaf 👋

A beta release will be available in a few hours.

You can start by downloading WebSSH on TestFlight here : https://testflight.apple.com/join/QSrBK59z

When the new release will be available you will receive a notification.

☀️ Have a nice day

**Reply by isontheline** [reply]
@bellaf WebSSH Beta 14.16.5 now available : https://testflight.apple.com/join/QSrBK59z

**Reply by isontheline** [reply]
@bellaf Have you got a try?

**Comment by bellaf** [comment]
Hi,

Yes, have had a go, it now picks up the ssh key files brilliantly, but doesn’t seem to want to connect to my servers, I am still playing about, the keys I’m using are 100% tested using other connections (putty etc etc)  so ought to work.  I am getting “ Provided credentials are bad error….

Still working on it, will let you if I find the problem ….  Unless you can advise there is another issue with using keys of course 😊 

Many thanks

Tony Bell

**Comment by bellaf** [comment]
Actually,

Been looking at it again…. All my keys are password protected,  so does web_ssh handle that password request?  Is it that that’s causing the rejection ?

Rgds

Tony Bell

**Reply by isontheline** [reply]
@bellaf Have you put the password of the key inside the form?

**Comment by bellaf** [comment]
Hi Arnaud,

Ah yes, I did spot that, and yes, the password for the key is there ..... yet it still rejects the connection

Cheers

Tony Bell

**Comment by bellaf** [comment]
Ok, think I might have solved it... I used an OpenSSh key to another server, and that works, so I suspect the puttygen generated key I was using isn’t right.... I will have a look and see if that clears the issue, otherwise 👍 all is good.
I’ll drop you a note when complete

Cheers

Tony Bell

**Reply by isontheline** [reply]
@bellaf Thank you a lot for your feedback Tony!

Happy to see that you can use WebSSH now 👍

Have a nice day ☀️
